File: cpp_version/Guarana.cpp
#include "Guarana.h"
#include "Swiat.h"
Guarana::Guarana(int x, int y, Swiat& swiat)
    : Roslina(0, x, y, swiat) {
}

char Guarana::rysowanie() const { return 'G'; }
std::string Guarana::getNazwa() const { return "Guarana"; }
std::shared_ptr<Organizm> Guarana::stworzPotomka(int x, int y) {
    return std::make_shared<Guarana>(x, y, swiat);
}

void Guarana::kolizja(Organizm& inny) {
    inny.zwiekszSile(3);
    swiat.dodajLog(inny.getNazwa() + " zjad³ Guaranê i zyska³ +3 si³y.");
    this->zabij();
}

void Guarana::zasiej(int nowyX, int nowyY) {
    swiat.dodajOrganizm(stworzPotomka(nowyX, nowyY));
}

File: java_version/Guarana.java
public class Guarana extends Roslina {
    public Guarana(int x, int y, Swiat swiat) {
        super(0, x, y, swiat);
    }

    @Override
    public char rysowanie() {
        return 'G';
    }

    @Override
    public String getNazwa() {
        return "Guarana";
    }

    @Override
    public Organizm stworzPotomka(int x, int y) {
        return new Guarana(x, y, swiat);
    }

    @Override
    public void kolizja(Organizm inny) {
        inny.zwiekszSile(3);
        swiat.dodajLog(inny.getNazwa() + " zjadł Guaranę i zyskał +3 siły.");
        this.zabij();
    }

    @Override
    protected void zasiej(int nowyX, int nowyY) {
        swiat.dodajOrganizm(stworzPotomka(nowyX, nowyY));
    }
}
